flink-ml-benchmark/src/main/java/org/apache/flink/ml/benchmark/BenchmarkResult.java

```diff
@@ -18,13 +18,25 @@
 
 package org.apache.flink.ml.benchmark;
 
+import org.apache.flink.ml.api.Stage;
+import org.apache.flink.ml.benchmark.datagenerator.DataGenerator;
+import org.apache.flink.ml.benchmark.datagenerator.InputDataGenerator;
 import org.apache.flink.util.Preconditions;
 
 /** The result of executing a benchmark. */
 public class BenchmarkResult {
     /** The benchmark name. */
     public final String name;
 
+    /** The stage to execute benchmark on. */
+    public final Stage<?> stage;
```

Comment:
Instead of keeping the stage/inputDataGenerator/modelDataGenerator instances in `BenchmarkResult`, would it be simpler to just pass the original `Map<String, ?> params` (which is used to instantiate these instances) to `BenchmarkUtils.getResultsMapAsJson(...)` directly?

Reply:
I added a `toMap` function to `BenchmarkResult` and a `ExceptionToMap` static method to `BenchmarkUtils`. Please see if this solution is good enough.

Reply:
Thanks for the update. The solution looks much better now.